packages/devtools/lib/src/settings/settings_controller.dart

```diff
@@ -2,28 +2,52 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'package:meta/meta.dart';
 import 'package:vm_service/vm_service.dart';
 
 import '../globals.dart';
+import '../service_registrations.dart' as registrations;
 import '../version.dart';
 
+// TODO(kenzie): we should listen for flag value updates and update the settings
+// screen with the new flag values. See
+// https://github.com/flutter/devtools/issues/988.
+
+typedef OnFlutterVersionChanged = void Function(FlutterVersion version);
+
+typedef OnFlagListChanged = void Function(FlagList flagList);
+
 class SettingsController {
-  SettingsController({this.onFlagListChange, this.onSdkVersionChange});
+  SettingsController({
+    @required this.onFlutterVersionChanged,
+    @required this.onFlagListChanged,
+  });
 
-  final Function(FlagList) onFlagListChange;
-  final Function(String) onSdkVersionChange;
+  final OnFlutterVersionChanged onFlutterVersionChanged;
 
-  Future<String> _getSdkVersion() async {
-    return serviceManager.sdkVersion;
-  }
+  final OnFlagListChanged onFlagListChanged;
 
-  Future<FlutterVersion> getFlutterVersion() async {
-    return FlutterVersion.parse(
-        (await serviceManager.getFlutterVersion()).json);
+  Future<void> entering() async {
+    onFlagListChanged(await serviceManager.service.getFlagList());
+    await _onFlutterVersionChanged();
   }
 
-  Future<void> entering() async {
-    onFlagListChange(await serviceManager.service.getFlagList());
-    onSdkVersionChange(await _getSdkVersion());
+  Future<void> _onFlutterVersionChanged() async {
+    if (await serviceManager.connectedApp.isAnyFlutterApp) {
+      serviceManager.hasRegisteredService(
+        registrations.flutterVersion.service,
+        (bool serviceAvailable) async {
+          if (serviceAvailable) {
+            final FlutterVersion version = FlutterVersion.parse(
+                (await serviceManager.getFlutterVersion()).json);
+            onFlutterVersionChanged(version);
```

Comment:
I think you probably want the Future<void> returned by this method to only complete when this callback is called the first time.
Likely doesn't matter much except for keeping tests from being flaky.